services/offer.js

```diff
@@ -1,14 +1,24 @@
 const Offer = require('~/models/offer')
+const userService = require('~/services/user')
 const Category = require('~/models/category')
 const Subject = require('~/models/subject')
 const { createError } = require('~/utils/errorsHelper')
 const { OFFER_NOT_FOUND, CATEGORY_NOT_FOUND, SUBJECT_NOT_FOUND } = require('~/consts/errors')
 
 const offerService = {
-  getOffers: async (match) => {
-    const offers = await Offer.find(match).lean().exec()
+  getOffers: async (match, sort, skip, limit) => {
+    const count = await Offer.countDocuments(match)
 
-    return offers
+    const offers = await Offer.find(match)
+      .populate({ path: 'authorId', select: ['totalReviews', '+photo'] })
+      .populate({ path: 'subjectId', select: 'name' })
+      .sort(sort)
+      .skip(skip)
+      .limit(limit)
+      .lean()
+      .exec()
+
+    return { offers, count }
```

Comment:
maybe it would be good to change variable naming **offers** to **items** like in **getUsers** service to follow some common pattern, how do you think about this idea?

Reply:
and maybe also change response body in swagger by adding count